rust/parquet/src/errors.rs

```diff
@@ -93,3 +93,14 @@ macro_rules! eof_err {
     ($fmt:expr) => (ParquetError::EOF($fmt.to_owned()));
     ($fmt:expr, $($args:expr),*) => (ParquetError::EOF(format!($fmt, $($args),*)));
 }
+
+// ----------------------------------------------------------------------
+// Some useful utility functions
+#[inline]
+pub fn check_in_range(index: usize, max_size: usize) -> Result<()> {
```

Comment:
Any reason this has to be in the `errors.rs` file? does it have to be in a function? it is only called in one place. Also we might want to panic directly since this seems unrecoverable...

Reply:
I put it here because I think it maybe a utility function so that other methods can use it. I think we should not panic directly and let application to handle it.